File: backend/app/utils/file_manager.py
import shutil
import time
from pathlib import Path

from app.config import settings


def create_task_dir(task_id: str) -> Path:
    task_dir = Path(settings.TEMP_DIR) / task_id
    task_dir.mkdir(parents=True, exist_ok=True)
    return task_dir


def get_task_dir(task_id: str) -> Path:
    return Path(settings.TEMP_DIR) / task_id


def cleanup_expired_tasks(max_age_hours: int = 24) -> None:
    temp_dir = Path(settings.TEMP_DIR)
    if not temp_dir.exists():
        return
    now = time.time()
    max_age_seconds = max_age_hours * 3600
    for task_dir in temp_dir.iterdir():
        if task_dir.is_dir():
            dir_mtime = task_dir.stat().st_mtime
            if now - dir_mtime > max_age_seconds:
                shutil.rmtree(task_dir, ignore_errors=True)

File: backend/src/main/java/com/bili/translator/util/FileManager.java
package com.bili.translator.util;

import com.bili.translator.config.AppProperties;
import org.springframework.stereotype.Component;

import java.io.File;
import java.nio.file.Path;
import java.nio.file.Paths;
import java.util.stream.Stream;

@Component
public class FileManager {

    private final AppProperties appProperties;

    public FileManager(AppProperties appProperties) {
        this.appProperties = appProperties;
        ensureTempDir();
    }

    private void ensureTempDir() {
        Path tempDir = Paths.get(appProperties.getTempDir());
        tempDir.toFile().mkdirs();
    }

    public Path createTaskDir(String taskId) {
        Path taskDir = Paths.get(appProperties.getTempDir(), taskId);
        taskDir.toFile().mkdirs();
        return taskDir;
    }

    public Path getTaskDir(String taskId) {
        return Paths.get(appProperties.getTempDir(), taskId);
    }

    public void cleanupExpiredTasks(int maxAgeHours) {
        File tempDir = Paths.get(appProperties.getTempDir()).toFile();
        if (!tempDir.exists() || !tempDir.isDirectory()) return;

        long now = System.currentTimeMillis();
        long maxAgeMillis = maxAgeHours * 3600L * 1000L;

        File[] dirs = tempDir.listFiles(File::isDirectory);
        if (dirs == null) return;

        for (File dir : dirs) {
            if (now - dir.lastModified() > maxAgeMillis) {
                deleteDirectory(dir);
            }
        }
    }

    private void deleteDirectory(File dir) {
        File[] files = dir.listFiles();
        if (files != null) {
            for (File f : files) {
                if (f.isDirectory()) deleteDirectory(f);
                else f.delete();
            }
        }
        dir.delete();
    }
}
